frontend/src/features/admin-form/settings/components/MultiLanguageSection/FormMultiLanguageToggle.tsx

```diff
@@ -0,0 +1,202 @@
+import { useCallback, useMemo, useState } from 'react'
+import { BiEditAlt } from 'react-icons/bi'
+import { GoEye, GoEyeClosed } from 'react-icons/go'
+import {
+  Box,
+  Divider,
+  Flex,
+  HStack,
+  IconButton,
+  Skeleton,
+  Text,
+} from '@chakra-ui/react'
+import _ from 'lodash'
+
+import { Language } from '~shared/types'
+
+import { convertUnicodeLocaleToLanguage } from '~utils/multiLanguage'
+import Badge from '~components/Badge'
+import Toggle from '~components/Toggle'
+
+import { useMutateFormSettings } from '../../mutations'
+import { useAdminFormSettings } from '../../queries'
+
+interface LanguageTranslationRowProps {
+  unicodeLocale: Language
+  isDefaultLanguage: boolean
+  isLast?: boolean
+}
+
+interface LanguageTranslationSectionProps {
+  defaultLanguage: Language
+}
+
+const LanguageTranslationRow = ({
+  unicodeLocale,
+  isDefaultLanguage,
+  isLast,
+}: LanguageTranslationRowProps): JSX.Element => {
+  const { data: settings } = useAdminFormSettings()
+
+  const supportedLanguages = settings?.supportedLanguages ?? null
+
+  const [isLanguageSupported, setIsLanguageSupported] = useState(
+    supportedLanguages?.indexOf(unicodeLocale) !== -1,
+  )
+
+  const languageToDisplay = convertUnicodeLocaleToLanguage(unicodeLocale)
+
+  const { mutateFormSupportedLanguages } = useMutateFormSettings()
+
+  const handleToggleSupportedLanague = useCallback(
+    (language: Language) => {
+      if (supportedLanguages == null) return
+
+      // Remove support for this language if it exists in supportedLanguages
```

Comment:
Suggestion: instead of using comments to describe `what` the code is doing, the code should probably be able to explain itself (i prefer reserving comments to describe `why` something is done)

The following code block (to line 76) could be replaced with something like this: 

```suggestion
      const isLanguageSupportedPreviously = !!supportedLanguages.find(language)
      const isAddSupport = !isLanguageSupportedPreviously
      const updatedSupportedLanguages = isAddSupport 
        ? supportedLanguage.concat(language)
        : supportedLanguages.filter(l => l !== language)
        
      setIsLanguageSupported(isAddSupport)
      
      return mutateFormSupportedLanguages.mutate({
          nextSupportedLanguages: updatedSupportedLanguages,
          selectedLanguage: language,
        })
```